src/create-glamorous.js

```diff
@@ -42,26 +45,29 @@ function createGlamorous(splitProps) {
        */
       const GlamorousComponent = withTheme(
         (props, context) => {
-          /* eslint no-use-before-define: 0 */
+          const updateClassName = shouldUpdate(props, context)
+
           const {toForward, cssOverrides, cssProp} = splitProps(
             props,
             GlamorousComponent,
           )
 
           // create className to apply
-          const className = getGlamorClassName({
-            styles: GlamorousComponent.styles,
-            props,
-            cssOverrides,
-            cssProp,
-            context,
-            displayName: GlamorousComponent.displayName,
-          })
+          GlamorousComponent.className = updateClassName ?
```

Comment:
The className and `GlamorousComponent.previous` you attach here is static to the GlamorousComponent, so any instance of the component will share the same className/previous. This seems problematic as it wouldn't support multiple usages of any component. For example: 

```
const MyComponent = glamorous.div(props => { color: props.color })

// Usage
<MyComponent color="green">green</MyComponent>
<MyComponent color="blue">blue</MyComponent>
```

When these two components re-render, the blue component's `shouldClassNameUpdate` function will be given: `prevProps.color` equal `green`, which isn't true, it used to be blue, and is still blue.